tests/capture/transforms/test_expand_plxpr_transforms.py

```diff
@@ -0,0 +1,292 @@
+
+
+class TestExpandTransformsInterpreter:
+    """Unit tests for ExpandTransformsInterpreter"""
+
+    def test_expand_transforms_interpreter_registration(self):
+        """Test that the primitives of PennyLane transforms are automatically registered with the
+        ExpandTransformsInterpreter."""
+
+        assert (
+            dummy_tape_only_transform._primitive
+            in ExpandTransformsInterpreter._primitive_registrations
+        )
+        assert (
+            dummy_tape_and_plxpr_transform._primitive
+            in ExpandTransformsInterpreter._primitive_registrations
+        )
+
+    def test_expand_transforms_interpreter_non_plxpr_transform(self):
+        """Test that transforms that do not have a valid ``plxpr_transform`` attribute
+        raise an error."""
+
+        custom_handler = ExpandTransformsInterpreter._primitive_registrations[
+            dummy_tape_only_transform._primitive
+        ]
+        assert dummy_tape_only_transform.plxpr_transform is None
+
+        dummy_interpreter = ExpandTransformsInterpreter()
+        dummy_invals = []
+        dummy_params = {
+            "inner_jaxpr": None,
+            "args_slice": None,
+            "consts_slice": None,
+            "targs_slice": None,
+            "tkwargs": {},
+        }
+
+        with pytest.raises(NotImplementedError):
+            custom_handler(dummy_interpreter, *dummy_invals, **dummy_params)
+
+    def test_expand_transforms_interpreter_plxpr_transform(self):
+        """Test that transforms that have a valid ``plxpr_transform`` are handled
+        correctly."""
+
+        custom_handler = ExpandTransformsInterpreter._primitive_registrations[
+            dummy_tape_and_plxpr_transform._primitive
+        ]
+        assert dummy_tape_and_plxpr_transform.plxpr_transform is not None
+
+        def f(x):
+            qml.RX(x, 0)
+            qml.CNOT([0, 1])
+            return qml.expval(qml.Z(0))
+
+        args = (1.5,)
+        jaxpr = jax.make_jaxpr(f)(*args)
+
+        def wrapper(*inner_args):
+            interpreter = ExpandTransformsInterpreter()
+            invals = [*inner_args, *jaxpr.consts]
+            params = {
+                "inner_jaxpr": jaxpr.jaxpr,
+                "args_slice": slice(0, len(inner_args)),
+                "consts_slice": slice(len(inner_args), len(jaxpr.consts) + len(inner_args)),
+                "targs_slice": slice(len(jaxpr.consts) + len(inner_args), None),
+                "tkwargs": {},
+            }
+            return custom_handler(interpreter, *invals, **params)
+
+        new_jaxpr = jax.make_jaxpr(wrapper)(*args)
+        assert all(
+            orig_eqn.primitive == new_eqn.primitive
+            for orig_eqn, new_eqn in zip(jaxpr.eqns, new_jaxpr.eqns, strict=True)
+        )
+        assert all(
+            orig_eqn.params == new_eqn.params
+            for orig_eqn, new_eqn in zip(jaxpr.eqns, new_jaxpr.eqns, strict=True)
+        )
+
+
+class TestExpandPlxprTransforms:
```

Comment:
Tests look good to me!

I would personally add also one test to ensure that `expand_plxpr_transforms` leaves the callable untouched if no transforms are present. For example

```
def f():
    qml.X(0)
    return qml.expval(qml.Z(0))

jaxpr = jax.make_jaxpr(f)()

transformed_f = expand_plxpr_transforms(f)
transformed_jaxpr = jax.make_jaxpr(transformed_f)()
transformed_jaxpr

# check that `transformed_jaxpr` and `jaxpr` are the same

```

and potentially one test to check a `NotImplementedError` is raised if we consider a transform that is still not implemented (for example `split_non_commuting`), unless this case is already covered by `test_expand_transforms_interpreter_non_plxpr_transform`.


As discussed with Christina, it would be *very useful* to have a way to determine if two `jaxpr` are the same for our purpose (ideally, changing the implementation of `qml.equal` to accept `jaxpr`s as well)